packages/app/src/components/Visualization/BlankCanvas.js

```diff
@@ -1,26 +1,39 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import { sGetLoadError } from '../../reducers/loadError';
+import i18n from '@dhis2/d2-i18n';
+import LoadingMask from '../../widgets/LoadingMask';
+import styles from './styles/BlankCanvas.style';
+import { sGetLoadError, sGetIsLoading } from '../../reducers/loader';
+import chartErrorImg from '../../assets/chart-error-graphic.png';
 
 export const visContainerId = 'visualization-container';
 export const defaultCanvasMessage = 'Visualization Canvas';
 
-export const BlankCanvas = ({ error }) => {
+export const BlankCanvas = ({ loading, error }) => {
     const message = error ? error : defaultCanvasMessage;
 
-    return (
-        <section className="canvas">
-            <div id={visContainerId}>
-                <div style={{ margin: 20 }}>
-                    <span>{message}</span>
-                </div>
+    let canvasContent = <p style={styles.text}>{message}</p>;
+    if (loading) {
+        canvasContent = <LoadingMask />;
+    } else if (error) {
+        canvasContent = (
+            <div>
+                <img src={chartErrorImg} alt={i18n.t('Chart error')} />
+                <p style={styles.text}>{message}</p>
```

Comment:
*minor* `{message}` should just be `{error}` so we can remove the ternary from line 13 above.